### DISCUSSION TITLE
nextjs waits (suspense enabled) react-query queryFn to return not rendering suspense fallback

### DISCUSSION THREAD

**Discussion by seonghyeonkimm** [vercel/next.js#36385] [Help] [answered]
I am not sure that I should ask this question here or react-query discussion. I am facing the issues like below

* when I am developing nextjs app on my local machine,
* I enabled queryClient suspense options
* when nextjs server renders page having useQuery, it does not render suspense fallback but waits it to return.

Is it working properly? I understood that reactQuery queryFn should be suspended on the server and nextjs should render suspense fallback and hydration should be done in the browser. Anyone who has solid understanding about this situation? Please help me to figure this situdation out well.

**Comment by icyJoseph** [comment, marked as answer]
I think, given that React 18 is a major release, and that the Suspense behaviour has changed, it is not particularly good at this time to loose sleep over it. The CSR behaviour and lazy loading components are on place, but the rest should come later on, and it will require the ecosystem of libraries to catch up.

Here's some links that made me arrive to the above:

- https://github.com/vercel/swr/issues/1898#issuecomment-1094426113
- https://github.com/reactjs/rfcs/blob/main/text/0213-suspense-in-react-18.md#new-feature-server-side-rendering-support-with-streaming
- https://github.com/reactwg/react-18/discussions/37

Basically, data fetching libraries in SSR, at the moment, data fetching libraries, are actually resolving the request. The streaming down partial HTML, is not ready just yet.

**Reply by seonghyeonkimm** [reply]
Thanks for your answer. I was using nextjs and react-query (suspense option enabled) until now. And it was working as I expected until now. So I thought it already supports suspense for data-fetching. I did not know that react itself is not supporting suspense for data-fetching. I should look into more specifically what I am confused about now.  Thanks again for helping me to figure this situation out.
